src/common/hooks/useShouldZeroApprove/shouldZeroApprove.ts

```diff
@@ -0,0 +1,44 @@
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import { Erc20 } from 'legacy/abis/types'
+import { ApprovalState } from 'legacy/hooks/useApproveCallback'
+
+interface ShouldZeroApproveParams {
+  approvalState: ApprovalState
+  tokenContract: Erc20 | null
+  spender: string | undefined
+  amountToApprove: CurrencyAmount<Currency> | undefined
+}
+
+export async function shouldZeroApprove({
+  approvalState,
+  tokenContract,
+  spender,
+  amountToApprove,
+}: ShouldZeroApproveParams) {
+  const shouldApprove = approvalState === ApprovalState.NOT_APPROVED || approvalState === ApprovalState.PENDING
+  if (tokenContract && spender && amountToApprove) {
```

Comment:
If you short-circuit this IF, you make the code simpler. Otherwise you create LOOONG ifs, and nested logic

Reply:
Addressed in https://github.com/cowprotocol/cowswap/pull/2529